notebook/static/notebook/js/notebook.js

```diff
@@ -1446,6 +1446,12 @@ define(function (require) {
         if (indices.length <= 1) {
             return;
         }
+
+        // Check if trying to merge above on topmost cell
+        if (indices.indexOf(-1) > -1) {
```

Comment:
Do we want to be more restrictive and check that no indices is negative ? (`.filter(function(value){return value < 0})`)